react/__mocks__/productMock.js

```diff
@@ -93,9 +95,12 @@ export const productMock = itemNumber => {
     },
   }
 
+  let customNameCount = 0
```

Comment:
You can use a more Functional Programming approach to create the items. I think using:
```Javascript
items: map(i =>  itemMock(`name-${i}`), range(1, itemNumber+1))
```
Would do the same and looks more concise.